core/src/main/java/org/apache/calcite/rel/rules/IntersectToDistinctRule.java

```diff
@@ -96,39 +91,48 @@ public IntersectToDistinctRule(Class<? extends Intersect> intersectClass,
     final RelOptCluster cluster = intersect.getCluster();
     final RexBuilder rexBuilder = cluster.getRexBuilder();
     final RelBuilder relBuilder = call.builder();
-
-    // 1st level GB: create a GB (col0, col1, count() as c) for each branch
-    for (RelNode input : intersect.getInputs()) {
-      relBuilder.push(input);
-      relBuilder.aggregate(relBuilder.groupKey(relBuilder.fields()),
-          relBuilder.countStar(null));
+    final String countPrefix = "count_i";
```

Comment:
You've removed more comments than you've added. This method doesn't need line-by-line comments but a 'here's the overall plan' comment is necessary. Give a good example of the SQL you intend to generate.